tests/system/src/test/java/test/com/sun/marlin/Scale0Test.java

```diff
@@ -0,0 +1,253 @@
+                setMinorTickCount(2);
+                setPrefWidth(200);
+                setShowTickLabels(true);
+                setShowTickMarks(true);
+            }
+        };
+        leftPane = new Group();
+
+        final NodeAndGraphic leftNode = create();
+        preparePane(this.leftPane, leftNode.node);
+
+        try {
+            String propertyName = "scaleXProperty"; //Works fine for translateXProperty
+            Method method = leftNode.graphic.getClass().getMethod(propertyName, (Class[]) null);
+            Object bindableObj = method.invoke(leftNode.graphic);
+            Method bindMethod = bindableObj.getClass().getMethod("bind", ObservableValue.class);
+            bindMethod.invoke(bindableObj, slider.valueProperty());
+        } catch (Throwable th) {
+            Logger.getLogger(Scale0Test.class.getName()).log(Level.SEVERE, "bind exception", th);
+        }
+
+        final Pane leftContainer = new Pane() {
+            {
+                setStyle("-fx-border-color: rosybrown;");
+                getChildren().add(leftPane);
+                setPrefSize(300, 300);
+                setMaxSize(300, 300);
+                setMinSize(300, 300);
+            }
+        };
+        GridPane.setConstraints(leftContainer, 0, 2);
+
+        GridPane field = new GridPane() {
+            {
+                getChildren().addAll(slider, leftContainer);
+            }
+        };
+
+        return new Scene(field, SIZE, SIZE, Color.WHITE);
+    }
+
+    private static NodeAndGraphic create() {
+        Button button = new Button();
+        button.setLayoutX(50);
+        button.setLayoutY(50);
+        button.setPrefSize(100, 50);
+        button.setMinSize(100, 50);
+        button.setMaxSize(100, 50);
+        if (button instanceof Labeled) {
+            Labeled l = (Labeled) button;
+            Circle circle = new Circle(10);
+            circle.setFill(Color.LIGHTGREEN);
+            circle.setStroke(Color.DARKGREEN);
+            circle.getStrokeDashArray().add(10.);
+            circle.getStrokeDashArray().add(8.);
+            l.setGraphic(circle);
+        }
+        return new NodeAndGraphic(button, button.getGraphic());
+    }
+
+    private static void preparePane(Group pane, Node node) {
+        pane.getChildren().clear();
+        final Rectangle bounds = new Rectangle() {
+            {
+                setWidth(300);
+                setHeight(300);
+                setFill(Color.TRANSPARENT);
+            }
+        };
+
+        pane.getChildren().add(bounds);
+        pane.setClip(new Rectangle() {
+            {
+                setWidth(300);
+                setHeight(300);
+            }
+        });
+        pane.getChildren().add(node);
+    }
+
+    final static class NodeAndGraphic {
+
+        final Node node;
+        final Object graphic;
+
+        NodeAndGraphic(Node node, Object graphic) {
+            this.node = node;
+            this.graphic = graphic;
+        }
+    }
+
+}
```

Comment:
I had few comments about the test like:
1. MyApp class can be removed
2. if (button instanceof Labeled) can be removed
3. preparePane() method could be shortened.
4. Variables leftPane and slider could be local
5. final keyword could be removed from some local variables.
6. Rename the test file and test function.

I made these changes locally and sharing the file as attached.

[ScaleX0Test.zip](https://github.com/openjdk/jfx/files/14123536/ScaleX0Test.zip)